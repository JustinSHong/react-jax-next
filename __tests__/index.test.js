import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Router from "next/router";

import Home from "../pages/index.js";

const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("with snapshot testing", () => {
    it("Home matches previous snapshot", () => {
        const component = renderer.create(<Home />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Home", () => {
    it("Should have a header", () => {
        const component = shallow(<Home />);
        expect(component.find("h1")).toHaveLength(1);
    });
});
