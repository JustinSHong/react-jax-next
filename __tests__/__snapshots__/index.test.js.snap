// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with snapshot testing Home matches previous snapshot 1`] = `
<div>
  <nav
    className="jsx-79800652"
  >
    <a
      className="jsx-79800652"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="jsx-79800652"
      href="/characters"
      onClick={[Function]}
    >
      Characters
    </a>
    <a
      className="jsx-79800652"
      href="/planets"
      onClick={[Function]}
    >
      Planets
    </a>
  </nav>
  <h1>
    My First Little Next.js App...featuring Star Wars!
  </h1>
</div>
`;
